posthog/temporal/batch_exports/bigquery_batch_export.py

```diff
@@ -174,48 +173,56 @@ async def adelete_table(
         project_id: str,
         dataset_id: str,
         table_id: str,
-        table_schema: list[bigquery.SchemaField],
         not_found_ok: bool = True,
     ) -> None:
         """Delete a table in BigQuery."""
         fully_qualified_name = f"{project_id}.{dataset_id}.{table_id}"
-        table = bigquery.Table(fully_qualified_name, schema=table_schema)
+        table = bigquery.Table(fully_qualified_name)
 
         await asyncio.to_thread(self.delete_table, table, not_found_ok=not_found_ok)
 
         return None
 
+    async def aget_table(
+        self,
+        project_id: str,
+        dataset_id: str,
+        table_id: str,
+    ) -> bigquery.Table:
+        """Get a table in BigQuery."""
+        fully_qualified_name = f"{project_id}.{dataset_id}.{table_id}"
+        return await asyncio.to_thread(self.get_table, fully_qualified_name)
+
     @contextlib.asynccontextmanager
     async def managed_table(
         self,
         project_id: str,
         dataset_id: str,
         table_id: str,
-        table_schema: list[bigquery.SchemaField],
+        table_schema: list[bigquery.SchemaField] | None = None,
         exists_ok: bool = True,
         not_found_ok: bool = True,
         delete: bool = True,
         create: bool = True,
     ) -> collections.abc.AsyncGenerator[bigquery.Table, None]:
         """Manage a table in BigQuery by ensure it exists while in context."""
         if create is True:
+            assert table_schema is not None, "Table schema is required when creating a table"
```

Comment:
Assertions are removed by optimized Python, I'm wondering if we would prefer to still keep the check by instead raising a `ValueError("Table schema is required when creating a table")` or other non-retryable error here.

Do we ever expect schema to be `None`?

Reply:
> Assertions are removed by optimized Python
ah right, I wasn't aware of this, thanks!

> Do we ever expect schema to be `None`?
To be honest, we're always called `managed_table` with a schema so we can probably remove this check